### PR TITLE
Further attempt at making the coverage report more realistic

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

File: .github/workflows/coveralls.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
     - name: setup hostname workaround
       run: sudo hostname localhost
     - name: config
-      run: CC=gcc ./config --debug --coverage ${{ matrix.branches.extra_config }} no-asm enable-rc5 enable-md2 enable-ssl3 enable-nextprotoneg enable-ssl3-method enable-weak-ssl-ciphers enable-zlib enable-ec_nistp_64_gcc_128 no-shared enable-buildtest-c++ enable-external-tests -DPEDANTIC -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+      run: CC=gcc ./config --debug --coverage ${{ matrix.branches.extra_config }} no-asm enable-rc5 enable-md2 enable-ssl3 enable-nextprotoneg enable-ssl3-method enable-weak-ssl-ciphers enable-zlib enable-ec_nistp_64_gcc_128 enable-buildtest-c++ enable-ssl-trace enable-trace
```
Comment (paulidale) [inline .github/workflows/coveralls.yml:59]: Why remove the fuzzing define?
It will cover additional places, albeit less important ones.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Of course the second commit is to be dropped when merging.

**Comment by paulidale** [comment]
Interesting that this improved things so much.

**Comment by t8m** [comment]
> Interesting that this improved things so much.

I am not surprised given for example apps/cmp.c was not previously shown as covered at all due to `-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION`. And the fips module code was not properly coverage-mapped either.

**Comment by t8m** [comment]
@paulidale still OK?

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Yep, still good.

**Comment by mattcaswell** [comment]
How beneficial is the sstraceref related change? I fear this will be a big burden. It is already a pain everytime something changes the ssltraceref output.

**Comment by t8m** [comment]
@mattcaswell it is the only thing to get some coverage of the ssl_trace along with the zlib. Alternatively I could drop the enable-zlib from the coveralls build.

**Comment by mattcaswell** [comment]
Hmmm. Neither option seems appealing. :-(

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
enable-tfo was lost in the fix, I've re-added it.

Assuming this is still OK and does not need restart of the 24h timeout.

**Review by mattcaswell — APPROVED** [review]
Still ok

**Comment by t8m** [comment]
Merged to the master branch. Thank you for the reviews.

**Pull request closed without merging**